File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.js
function solution(numbers, target) {
  function dfs(index, sum) {
    // 다 돌았을 때 합이 target과 같으면 1, 아니면 0을 반환
    if (index === numbers.length) {
      return sum === target ? 1 : 0;
    }
    
    const plus = dfs(index + 1, sum + numbers[index]);
    const minus = dfs(index + 1, sum - numbers[index]);

    return plus + minus;
  }

  return dfs(0, 0);
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    answer = 0
    arr = [0]
    for num in numbers:
        tmp = []
        for parent in arr:
            tmp.append(parent + num)
            tmp.append(parent - num)
        arr = tmp
    for leaf in arr:
        if leaf == target:
            answer += 1
    return answer